### PR TITLE
xref zip file loading order

### PR DESCRIPTION
**Pull request by live1206**
https://ceapex.visualstudio.com/Engineering/_workitems/edit/118436

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5072)

### CODE REVIEW

**Comment by mingwli** [comment]
>                 foreach (var entry in archive.Entries)

is order stable here?

---
Refers to: src/docfx/build/xref/ExternalXrefMapLoader.cs:76 in fe43f0d. [](commit_id = fe43f0d683253658d216e7442472c6c652bc4596, deletion_comment = False)

**Comment by live1206** [comment]
> > ```
> >             foreach (var entry in archive.Entries)
> > ```
> 
> is order stable here?
> 
> Refers to: src/docfx/build/xref/ExternalXrefMapLoader.cs:76 in [fe43f0d](https://github.com/dotnet/docfx/commit/fe43f0d683253658d216e7442472c6c652bc4596). [](commit_id = [fe43f0d](https://github.com/dotnet/docfx/commit/fe43f0d683253658d216e7442472c6c652bc4596), deletion_comment = False)

I think so since the type of `arvhive.Entries` is `ReadOnlyCollection<ZipArchiveEntry>`

File: src/docfx/build/xref/ExternalXrefMapLoader.cs [hunk 1]
```diff
@@ -82,15 +82,15 @@ private static void LoadZipFile(Dictionary<string, Lazy<ExternalXrefSpec>> resul
                             var xrefMap = YamlUtility.Deserialize<XrefMapModel>(sr.ReadToEnd(), new FilePath(url));
                             foreach (var spec in xrefMap.References)
                             {
-                                result.TryAdd(spec.Uid, new Lazy<ExternalXrefSpec>(() => spec));
+                                result[spec.Uid] = new Lazy<ExternalXrefSpec>(() => spec);
```
Comment (OsmondJiang) [inline src/docfx/build/xref/ExternalXrefMapLoader.cs]: this means the latter one has higher priority, does it apply to all or only works in zip files?

Comment (OsmondJiang) [reply src/docfx/build/xref/ExternalXrefMapLoader.cs]: for the other parts, we are still using TryAdd

Comment (live1206) [reply src/docfx/build/xref/ExternalXrefMapLoader.cs]: It only applies for zip files as v2 does. And yes, the behavior is inconsistent in v2

Comment (OsmondJiang) [reply src/docfx/build/xref/ExternalXrefMapLoader.cs]: great to know

**Comment by live1206** [comment]
#4450

**Pull request merged**